Missing 'uv' dev dependency

**Steps to Reproduce**

```
git clone git@github.com:app-sre/rh-aws-saml-login.git
cd rh-aws-saml-login
python3.11 -m venv .venv
source .venv/bin/activate
pip install -e '.[dev]'
make test
```

**Current Behavior**

```
uv run --frozen ruff check --no-fix
make: uv: No such file or directory
make: *** [Makefile:11: test] Error 127
```

**Expected Behavior**

`make test` runs linters and tests and exits with return code `0`.


**Additional Info**

I would prepare a patch adding `uv` to the dev dependencies, but:

1. This package defines minimal required versions for all dependencies.
2. `uv` has a very rapid release cycle.
3. I don't know what minimal version is safe. A version too old might not work. A version too recent may cause issues in existing environments, considering that `uv` must have been installed separately in those environments.

--- a/README.md
+++ b/README.md
@@ -195,6 +195,17 @@ Opens the AWS web console for the `s3` service in the `app-sre` account.
 
 ## Development
 
+`rh-aws-saml-login` uses [uv](https://github.com/astral-sh/uv) for project and dependency management. Follow the [uv installation instructions](https://docs.astral.sh/uv/getting-started/installation/) to install it in on your local machine.
+
+### Setup
+
+1. Fork the repository and clone it
+1. Create a Python development environment with `make dev-venv`
+1. Activate the virtual environment with `source .venv/bin/activate`
+1. Happy coding!
+
+### Release
+
 - Update CHANGELOG.md with the new version number and date
 - Bump the version number in [pyproject.toml](/pyproject.toml)
 

